test/view_models/paye/PayeYourTaxableIncomeSpec.scala

```diff
@@ -0,0 +1,124 @@
+    payload = Some(Map(
+      "self_employment_income" -> Amount(100, "GBP"),
+      "income_from_employment" -> Amount(200, "GBP"),
+      "state_pension" -> Amount(300, "GBP"),
+      "other_pension_income" -> Amount(400, "GBP"),
+      "taxable_state_benefits" -> Amount(500, "GBP"),
+      "other_income" -> Amount(600, "GBP"),
+      "benefits_from_employment" -> Amount(700, "GBP"),
+      "total_income_before_tax" -> Amount(800, "GBP")
+    )),
+    rates = None,
+    incomeTaxStatus = None
+  ))
+
+  val incomeTaxDataStatePensionNoIncomeFromEmployment = Some(DataHolder(
+    payload = Some(Map(
+      "self_employment_income" -> Amount(100, "GBP"),
+      "income_from_employment" -> Amount(0, "GBP"),
+      "state_pension" -> Amount(300, "GBP"),
+      "other_pension_income" -> Amount(400, "GBP"),
+      "taxable_state_benefits" -> Amount(500, "GBP"),
+      "other_income" -> Amount(600, "GBP"),
+      "benefits_from_employment" -> Amount(700, "GBP"),
+      "total_income_before_tax" -> Amount(800, "GBP")
+      )),
+    rates = None,
+    incomeTaxStatus = None
+    ))
+
+  val IncomeTaxJustOtherPension = Some(DataHolder(
+    payload = Some(Map(
+      "self_employment_income" -> Amount(100, "GBP"),
+      "income_from_employment" -> Amount(200, "GBP"),
+      "other_pension_income" -> Amount(400, "GBP"),
+      "taxable_state_benefits" -> Amount(500, "GBP"),
+      "other_income" -> Amount(600, "GBP"),
+      "benefits_from_employment" -> Amount(700, "GBP"),
+      "total_income_before_tax" -> Amount(800, "GBP")
+      )),
+    rates = None,
+    incomeTaxStatus = None
+    ))
+  
+  def incomeTaxPayeAtsData(incomeTax: Option[DataHolder]): PayeAtsData = {
+      PayeAtsData(2018, None, None, incomeTax, None, None)
+    }
+  
+    "PayeYourTaxableIncome" should {
+      "Transform income tax data just other pension to view model" in {
+        val expectedIncomeeTaxRows = List(IncomeTaxRow("self_employment_income", Amount(100, "GBP")),
+                                          IncomeTaxRow("income_from_employment", Amount(200, "GBP")),
+                                          IncomeTaxRow("state_pension", Amount(300, "GBP")),
+                                          IncomeTaxRow("other_pension_income", Amount(400, "GBP")),
+                                          IncomeTaxRow("taxable_state_benefits", Amount(500, "GBP")),
+                                          IncomeTaxRow("other_income", Amount(600, "GBP")),
+                                          IncomeTaxRow("benefits_from_employment", Amount(700, "GBP")))
+        val expectedIncomeBeforeTaxTotal = Amount(800, "GBP")
+
+        val payeAtsData = incomeTaxPayeAtsData(incomeTaxDataStatePensionAndOther)
+        val viewModel = PayeYourTaxableIncome.buildViewModel(payeAtsData)
+
+        viewModel.get.incomeTaxRows shouldBe (expectedIncomeeTaxRows)
+        viewModel.get.incomeBeforeTaxTaxTotal shouldBe expectedIncomeBeforeTaxTotal
```

Comment:
Don't use .get, prefer` shouldBe Some(expected)` as it does not throw

Reply:
Agreed. Done 👍